src/XML/src/FMI3/fmi3_xml_type.c

```diff
@@ -637,15 +700,21 @@ int fmi3_xml_handle_SimpleType(fmi3_xml_parser_context_t *context, const char* d
  * correct type and make sure that enough memory is given via parameter 'typeSize'.
  */
 fmi3_xml_variable_type_base_t* fmi3_xml_alloc_variable_or_typedef_props(fmi3_xml_type_definitions_t* td,
-        fmi3_xml_variable_type_base_t* base, size_t typeSize)
+        fmi3_xml_variable_type_base_t* nextLayer, size_t typeSize)
 {
     jm_callbacks* cb = td->typeDefinitions.callbacks;
     fmi3_xml_variable_type_base_t* type = cb->malloc(typeSize);
-    if (!type) return NULL;
-    fmi3_xml_init_variable_type_base(type, fmi3_xml_type_struct_enu_props, base->baseType);
-    type->nextLayer = base;
+    if (!type) {
+        jm_log_fatal(cb, module, "Could not allocate memory"); // TODO: Change to fmi3_xml_parse_fatal
```

Comment:
Would cause quite some conflicts with array ticket if I did this TODO now, because it requires 'context'.

(I also want to rename this function. It does a lot more than just allocates.)